base/logging.jl

```diff
@@ -347,6 +307,50 @@ function logmsg_code(_module, file, line, level, message, exs...)
     end
 end
 
+function process_logmsg_exs(_module, _file, _line, level, message, exs...)
+    log_data = (;_module, _file, _line, kwargs=Any[])
+    for ex in exs
+        if ex isa Expr && ex.head === :(=) && ex.args[1] isa Symbol
+            k,v = ex.args
+            if !(k isa Symbol)
+                throw(ArgumentError("Expected symbol for key in key value pair `$ex`"))
+            end
+
+            # Recognize several special keyword arguments
+            if k ∈ (:_group, :_id, :_module, :_file, :_line)
+                log_data = Base.setindex(log_data, esc(v), k)
```

Comment:
I think this is a reasonable enough use of `setindex`.
Like worse case senario is it copies at 4+5+6+7+8 things if everything is overridden.
and this only happens during the macro expansion anyway.

I think this is clearer than having 10 lines of `if,elseif...` because it is clear in these 2 lines that they are all special in the same way .

Reply:
I'm not really sure this is an improvement in readability with having to use `setindex` like this. I mean I guess we could go with it, but overall I think it was quite straightforward before and extremely easy to see that all these five cases were the same. More verbose than unreadable ;-)

On the other hand bundling all these together into `log_data` at the bottom of process_logmsg_exs seems neat.

Reply:
Yeah I guess I can just flip the order and do it that way